view/multi_match.js

```diff
@@ -0,0 +1,43 @@
+/**
+  this view is wrapped in a function so it can be re-used
+**/
+
+module.exports = function( properties ){
+  return function( vs ){
+
+    // base view
+    var view = { multi_match: {} };
+
+    properties.forEach(function(property) {
+      // validate required params
+      if( !property ||
+          !vs.isset('input:'+property) ||
+          !vs.isset('admin:'+property+':field') ){
+               return null;
+      }
+    });
+
+    if (!vs.isset('input:'+properties[0])) {
+      return null;
+    }
+
+    view.multi_match.analyzer = 'peliasAdmin';
+    // TODO: do something better than just take the first non-empty property
+    var possibleValues = properties.map(function(property) {
+      return vs.var('input:'+property);
+    });
+
+    view.multi_match.query = possibleValues.find(function(value) {
+      return value !== null;
+    });
+
+    var fields = properties.map(function(property) {
+      var boost = vs.var('admin:'+property+':boost') || 1;
+      return vs.var('admin:'+property+':field') +'^' + boost;
```

Comment:
this also deserves a comment, considering how cryptic the ES syntax is for this.